src/classes/ERR_Handler.cls

```diff
@@ -402,6 +402,18 @@ public class ERR_Handler {
         processErrorsOnly(new List<Error__c>{ error }, context);
     }
 
+    /***
+    * @description returns true if this method is calling from data migration
+    * or validate migration context
+    * @returns Boolean
+    */
+    public static Boolean isValidateOrDataMigration(String context) {
```

Comment:
Nice solution!
To be on the safe side, we would need to check that context is not null, ie
 ````
return String.isNotBlank(context) 
    && (
        context.contains(RD2_DataMigrationBase_BATCH.LOG_CONTEXT_DRY_RUN)
        || context.contains(RD2_DataMigrationBase_BATCH.LOG_CONTEXT_MIGRATION)
     );
```

Suggestion for the name: `isMigrationContext()`  or `isRD2MigrationContext()`?